modules/core/04-channel/keeper/upgrade_test.go

```diff
@@ -431,7 +429,14 @@ func (suite *KeeperTestSuite) TestChanUpgradeTry() {
 				proofChannel, proofUpgradeTimeout, proofUpgradeSequence, proofHeight,
 			)
 
-			suite.chainB.GetSimApp().GetIBCKeeper().ChannelKeeper.WriteUpgradeTryChannel(suite.chainB.GetContext(), path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID, upgradeChannel.Version, upgradeSequence, upgradeChannel)
+			// mock what is happening in the msg server handler.
```

Comment:
Why is this needed?

Reply:
this is needed as the current tests are testing the channel keeper function and not the ibckeeper function. The error receipt is written in the ibc keeper. An alternative is to test the ibckeeper instead, and pass in a MsgChannelUpgradeTry and this would not be needed.